packages/cli/src/lib/build-strategies/strategies/ImageStrategy.ts

```diff
@@ -0,0 +1,330 @@
+import {
+  displayPath,
+  ensureDockerDaemonRunning,
+  FileLock,
+  generateDockerImageName,
+  isDockerInstalled,
+  runCommand,
+} from "../../system";
+import { BuildStrategyArgs, BuildStrategy } from "../BuildStrategy";
+import { intlMsg } from "../../intl";
+import { withSpinner } from "../../helpers";
+
+import fs from "fs";
+import path from "path";
+import { isWin, writeFileSync } from "@polywrap/os-js";
+import Mustache from "mustache";
+
+type BuildImageId = string;
+
+export class ImageBuildStrategy extends BuildStrategy<BuildImageId> {
+  private _dockerLock: FileLock;
+
+  constructor(args: BuildStrategyArgs) {
+    super(args);
+
+    if (!isDockerInstalled()) {
+      throw new Error(intlMsg.lib_docker_noInstall());
+    }
+
+    this._dockerLock = new FileLock(
+      this.project.getCachePath("build/DOCKER_LOCK"),
+      (msg) => {
+        throw new Error(msg);
+      }
+    );
+  }
+
+  getStrategyName(): string {
+    return "image";
+  }
+
+  public async buildSources(): Promise<BuildImageId> {
+    await this._dockerLock.request();
+    try {
+      await ensureDockerDaemonRunning();
+
+      const buildManifestPath = await this.project.getBuildManifestPath();
+      const buildManifestDir =
+        buildManifestPath && path.dirname(buildManifestPath);
+      const buildManifest = await this.project.getBuildManifest();
+      const imageName =
+        buildManifest?.strategies?.image?.name ||
+        generateDockerImageName(await this.project.getBuildUuid());
+
+      const language = await this.project.getManifestLanguage();
+
+      const dockerfileTemplatePath = path.join(
+        __dirname,
+        "..",
+        "..",
+        "defaults",
+        "build-strategies",
+        language,
+        this.getStrategyName(),
+        "Dockerfile.mustache"
+      );
+
+      let dockerfilePath: string;
+
+      const customManifestDockerfilePath =
+        buildManifest?.strategies?.image?.dockerfile;
+
+      if (buildManifestDir && customManifestDockerfilePath) {
+        dockerfilePath = path.join(
+          buildManifestDir,
+          customManifestDockerfilePath
+        );
+      } else {
+        dockerfilePath = this._generateDockerfile(dockerfileTemplatePath, {
+          ...buildManifest.config,
+          ...buildManifest.strategies?.image,
+        });
+      }
+
+      await this.project.cacheBuildManifestLinkedPackages();
+
+      const dockerBuildxConfig = buildManifest?.strategies?.image?.buildx;
+      const useBuildx = !!dockerBuildxConfig;
+
+      let cacheDir: string | undefined;
+      let removeBuilder = false;
+
+      if (dockerBuildxConfig && typeof dockerBuildxConfig !== "boolean") {
+        const cache = dockerBuildxConfig.cache;
+
+        if (cache == true) {
+          cacheDir = this.project.getCachePath("build/image/cache");
+        } else if (cache) {
+          if (!path.isAbsolute(cache)) {
```

Comment:
Just from a style perspective, maybe omit the invert the if so that we don't have to read the `!` which might be hard to notice at a cursory glance? Nitpicking here...

Reply:
Yep, I agree with this